docsk8s/administering/ls-k8s-stack-monitoring.asciidoc

```diff
@@ -1,14 +1,126 @@
 
-An important step to making your environment production ready is to configure stack monitoring. Monitoring logs and metrics data can be sent to an external resource, such as {ess} or {eck}, so that in the event that any components of your environment become unresponsive, your monitoring data is available.
+To enable {logstash-ref}/monitoring-with-metricbeat.html[Stack monitoring] for {ls}, you need {metricbeat} to collect {ls} metrics, {es} to store the metrics and {kib} to view the result.
 
-Follow these steps to configure monitoring:
+[[monitor-with-ECK]]
+==== Stack monitoring with Elastic Cloud on {k8s} (ECK)
 
-. Step 1
-. Step 2
-. Step 3
\ No newline at end of file
+TIP: Be sure that you have ECK installed so that you can follow the example. Check out <<qs-set-up>> for set up info.
+
+For these examples, we will be modifying the Beats stack monitoring link:https://github.com/elastic/cloud-on-k8s/blob/main/config/recipes/beats/stack_monitoring.yaml[recipe] from the ECK examples.
+This example initiates a production {es} cluster, a monitoring {es} cluster, {filebeat}, {metricbeat}, a production Kibana and a monitoring Kibana. It monitors {es} and Kibana and sends metrics to the monitoring cluster.
+We use {metricbeat-ref}/configuration-autodiscover.html[autodiscover] to configure monitoring for multiple {ls} instances.
+
+* <<ls-k8s-monitor-config-metricbeat>>
+* <<ls-k8s-monitor-config-ls>>
+* <<ls-k8s-monitor-kibana>>
+
+[float]
+[[ls-k8s-monitor-config-metricbeat]]
+===== Configure Metricbeat
+
+To monitor {ls}, add the `Logstash` module to the recipe.
+
+[source,yaml]
+--
+apiVersion: beat.k8s.elastic.co/v1beta1
+kind: Beat
+metadata:
+  name: metricbeat
+spec:
+  type: metricbeat
+  version: 8.4.2
+  elasticsearchRef:
+    name: elasticsearch-monitoring <1>
+  config:
+    metricbeat:
+      autodiscover:
+        providers:
+          - type: kubernetes
+            scope: cluster
+            hints.enabled: true
+            templates:
+              - condition:
+                  contains:
+                    kubernetes.labels.app: ls <2>
+                config:
+                  - module: logstash <3>
+                    metricsets:
+                      - node
+                      - node_stats
+                    period: 10s
+                    hosts: "http://${data.host}:9600"
+                    xpack.enabled: true
+...
+--
+
+<1> {metricbeat} sends metrics to `elasticsearch-monitoring` cluster.
+<2> {metricbeat} scans for the pods with label `app: ls` to collect {ls} metrics.
+<3> {metricbeat} logstash module calls metric endpoint of each {ls} from port `9600` for every `10` seconds.
+
+[float]
+[[ls-k8s-monitor-config-ls]]
+===== Configure {ls}
+
+Add label `app: ls` to `Deployment` for autodiscover.
```

Comment:
Do we need to mention `StatefulSet` here too?

Reply:
I think it is optional